plugins/modules/modprobe.py

```diff
@@ -154,6 +287,12 @@ def main():
     elif modprobe.desired_state == 'absent' and modprobe.module_loaded():
         modprobe.unload_module()
 
+    if modprobe.persistent:
+        if modprobe.desired_state == 'present' and not (modprobe.module_is_loaded_persistently and modprobe.params_is_set):
+            modprobe.load_module_permanent()
+        elif modprobe.desired_state == 'absent' and (modprobe.module_is_loaded_persistently or modprobe.permanent_params):
+            modprobe.unload_module_permanent()
+
```

Comment:
This means that `persistent=false` has absolutely no effect on the module. Not sure that is the desired behaviour, but even if it is, that should be better explained in the parameter description.

Another way would be to implement Felix's suggestion and create new states.

Reply:
Yes that is desired behavior.

1. Felix's suggestion has one major flaw: too many states needed to be added. That can be quite confusing. And we also have to come up with names for 6 desirable states: 
  a. Presented and not persistent 
  b. Presented and persistent 
  c. Absent and persistent 
  d. Absent and not persistent 
  e. Presented but with persistence set to absent(in case you want module to me activated now but to be disabled on next reboot 
  f. Absent but with persistence set to present(in case you want module to be disabled now but enable it on next reboot
2. Current code has another flaw: it can't implement states e and f.
3. As I mentioned previously we could use ```persistent``` option as multiple choice parameter with 3 options and this implements all of 6 states.

So here you have 3 ideas. I personally like third one, cause it has no flaws. But if you think that Felix's idea is better I could do that.

Reply:
So @russoz, what do you say?

Reply:
I decided to go with my idea, so I've changed persistent parameter to multiple choice str with 3 options:

- present - module persistently enabled
- absent - module persistently disabled
- disabled - persistent option doesn't do anything. You can use it if you don't care about persistency. 

Maybe naming could be better, but I can change it, if needed. Feel free to propose naming ideas.